pkg/services/publicdashboards/api/api.go

```diff
@@ -104,24 +109,66 @@ func (api *Api) GetPublicDashboard(c *models.ReqContext) response.Response {
 	}
 
 	pdc, err := api.PublicDashboardService.FindByDashboardUid(c.Req.Context(), c.OrgID, web.Params(c.Req)[":dashboardUid"])
+
 	if err != nil {
-		return api.handleError(c.Req.Context(), http.StatusInternalServerError, "GetPublicDashboardConfig: failed to get public dashboard config", err)
+		return api.handleError(c.Req.Context(), http.StatusInternalServerError, "GetPublicDashboard: failed to get public dashboard ", err)
+	}
+
+	if pdc == nil {
+		return api.handleError(c.Req.Context(), http.StatusNotFound, "GetPublicDashboard: public dashboard not found", ErrPublicDashboardNotFound)
```

Comment:
look at the service. If pubdash is nil, we return` (nil, ErrPublicDashboardNotFound).`
So, this code will never be executed, because error won't be null, therefore it will end up inside the other if statement.
I think it's ok the service to return that error. Also take into account that handleError method httpStatus and message params are defaults, so it should return ErrPublicDashboardNotFound in a good way

Reply:
Also, this is going to change with the new Grafana type errors so don't worry about `handleError`.

Reply:
Going to leave this for now since we will resolve with error refactor

Reply:
I would remove it anyway, because the refactor should be in err != nil if statement, and this should be removed. The one who makes the change to Grafana errors, may not be aware of removing this piece of code